cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,44 @@
 /// <reference types='cypress' />
 
 describe('Student Registration page', () => {
-  before(() => {
+  let user;
 
+  before(() => {
+    cy.visit('https://demoqa.com/automation-practice-form');
+    cy.task('generateUser').then((generateUser) => {
+      user= generateUser;
+    });
   });
 
-  it('', () => {
+  it('should register a new student', () => {
+    cy.findByPlaceholder('First Name').type(user.firstName);
+    cy.findByPlaceholder('Last Name').type(user.lastName);
+    cy.findByPlaceholder('name@example.com').type(user.email);
+
+    cy.get(`#gender-radio-${user.randomGenderIndex}`).click({ force: true });
+    cy.findByPlaceholder('Mobile Number').type(user.mobileNumber)
+
+    cy.get('#dateOfBirthInput').click();
+    cy.pickDate('month-select').select(user.birth.month);
+    cy.pickDate('year-select').select(`${user.birth.year}`);
+    cy.pickDate('day').contains(user.birth.day).click();
+
+    cy.get('.subjects-auto-complete__value-container')
+    .type('en{enter}' + 'ph{enter}');
+
+    cy.get('.custom-control-label').contains(user.hobby).click();
+
+    cy.findByPlaceholder('Current Address').type(user.address);
+    cy.get('#state').type('{downarrow}{enter}');
+    cy.get('#city').type('{downarrow}{enter}');
+
+    cy.get('#submit').click();
+
+    cy.contains('tr', 'Student Name')
```

Comment:
you have filled the form with a lof of values but assert only 3. You have to the check that the final result contains the selected values.